chain/ethereum/src/runtime/runtime_adapter.rs

```diff
@@ -43,13 +43,17 @@ impl blockchain::RuntimeAdapter<Chain> for RuntimeAdapter {
     fn host_fns(&self, ds: &DataSource) -> Result<Vec<HostFn>, Error> {
         let abis = ds.mapping.abis.clone();
         let call_cache = self.call_cache.cheap_clone();
-        let eth_adapter = self
-            .eth_adapters
-            .cheapest_with(&NodeCapabilities {
-                archive: ds.mapping.requires_archive()?,
-                traces: false,
-            })?
-            .cheap_clone();
+        let eth_adapter = match self.eth_adapters.call_only_adapter() {
+            Err(err) => bail!(err),
+            Ok(None) => self
+                .eth_adapters
+                .cheapest_with(&NodeCapabilities {
+                    archive: ds.mapping.requires_archive()?,
+                    traces: false,
+                })?
+                .cheap_clone(),
```

Comment:
It would be better to have this logic (first look for call only adapters, then look in the general adapters) encapsulated in a method on `eth_adapters`, maybe `eth_adapters.cheapest_for_call(..)` since this whole `match` statement has a lot of knowledge about how `EthereumNetworkAdapters` sets itself up internally which would be better to hide away

Reply:
good catch, I had added the function later, forgot to replace this part :)